src/internalEnforcer.ts

```diff
@@ -24,7 +24,7 @@ export class InternalEnforcer extends CoreEnforcer {
   /**
    * addPolicyInternal adds a rule to the current policy.
    */
-  public async addPolicyInternal(sec: string, ptype: string, rule: string[]): Promise<boolean> {
+  protected async addPolicyInternal(sec: string, ptype: string, rule: string[]): Promise<boolean> {
     if (this.model.hasPolicy(sec, ptype, rule)) {
```

Comment:
I think these function could just use `WithoutNotify` functions. There is quite a lot of code repetition.

Reply:
I thought of going with go-casbin's implementation, but I think we should use Without Notify in these methods and call watcher

Reply:
I think we add a param to decide whether call the watcher on the `xxPolicyInternal()` method.

```typescript
protected async addPolicyInternal(useWatcher: bool, sec: string, ptype: string, rule: string[]): Promise<boolean> {

}
```